csharp/ql/lib/semmle/code/csharp/Generics.qll

```diff
@@ -261,7 +261,11 @@ class TypeParameter extends DotNet::TypeParameter, Type, @type_parameter {
  */
 class TypeParameterConstraints extends Element, @type_parameter_constraints {
   /** Gets a specific type constraint, if any. */
-  Type getATypeConstraint() { specific_type_parameter_constraints(this, getTypeRef(result)) }
+  Type getATypeConstraint() {
+    specific_type_parameter_constraints(this, result)
+    or
+    specific_type_parameter_constraints(this, getTypeRef(result))
```

Comment:
Is it intentional that there is no negation here as well?

Reply:
I suppose this because you need to preserve the semantics of the original code in this case (as reflexivity was removed from `getTypeRef`.)

Reply:
We can only apply the negation when the type is supposed to be unique (such as the type of an expression), but not when there can be multiple types (such as implemented interfaces or, as in the case above, type parameter constraints).